docs/how_to/use_optimizationlevel.rst

```diff
@@ -0,0 +1,306 @@
+    «     q0_2 -> 1 ┤ Rz(-0.75949) ├┤ √X ├┤ Rz(π/4) ├┤ X ├┤ Rz(π/4) ├┤ X ├»
+    «               └──────────────┘└────┘└─────────┘└─┬─┘└─────────┘└───┘»
+    «     q0_0 -> 2 ───────────────────────────────────■──────────────────»
+    «                                                                     »
+    «ancilla_0 -> 3 ──────────────────────────────────────────────────────»
+    «                                                                     »
+    «ancilla_1 -> 4 ──────────────────────────────────────────────────────»
+    «                                                                     »
+    «               ┌─────────┐                                        ┌───┐     »
+    «     q0_1 -> 0 ┤ Rz(π/4) ├─────────────────────────────────────■──┤ X ├──■──»
+    «               ├─────────┴┐┌───┐┌──────────┐┌────┐┌─────────┐┌─┴─┐└─┬─┘┌─┴─┐»
+    «     q0_2 -> 1 ┤ Rz(-π/4) ├┤ X ├┤ Rz(3π/4) ├┤ √X ├┤ Rz(π/2) ├┤ X ├──■──┤ X ├»
+    «               └──────────┘└─┬─┘└──────────┘└────┘└─────────┘└───┘     └───┘»
+    «     q0_0 -> 2 ──────────────■──────────────────────────────────────────────»
+    «                                                                            »
+    «ancilla_0 -> 3 ─────────────────────────────────────────────────────────────»
+    «                                                                            »
+    «ancilla_1 -> 4 ─────────────────────────────────────────────────────────────»
+    «                                                                            »
+    «                                          
+    «     q0_1 -> 0 ────────────────────────■──
+    «               ┌───┐┌──────────┐┌───┐┌─┴─┐
+    «     q0_2 -> 1 ┤ X ├┤ Rz(-π/4) ├┤ X ├┤ X ├
+    «               └─┬─┘├─────────┬┘└─┬─┘└───┘
+    «     q0_0 -> 2 ──■──┤ Rz(π/4) ├───■───────
+    «                    └─────────┘           
+    «ancilla_0 -> 3 ───────────────────────────
+    «                                          
+    «ancilla_1 -> 4 ───────────────────────────
+    «                                          
+
+
+Plotting the Results
+====================
+
+You can visualize the results of your previous examples by generating plot that show the depth, number of gates, and number of CX gates of your quantum circuits.
+
+.. testcode::
+
+    
+    import matplotlib.pyplot as plt # import matplotlib to plot your result
+    
+    
+    fig, ax = plt.subplots()
+    my_xticks = [str(i) for i in range(4)]
+    plt.xticks(range(4), my_xticks)
+    ax.plot(range(4), [qc_b0.depth(),qc_b1.depth(),qc_b2.depth(),qc_b3.depth()],label = "Number of depth", marker='o',color ='#6929C4')
+    ax.plot(range(4), [qc_b0.size(),qc_B1.size(),qc_b2.size(),qc_b3.size()],label = "Number of gates", marker='o',color ='blue')
+    ax.plot(range(4), [qc_b0.num_nonlocal_gates(),qc_b1.num_nonlocal_gates(),qc_b2.num_nonlocal_gates(),qc_b3.num_nonlocal_gates()],label = "Number of non local gates", marker='o',color ='green')
```

Comment:
This is hard to read. It would be better to format this over several lines. Tip: copy the code into a file, then run `black` on it. For example, I like to use the tool `pipx` to install Black globally, then you can run `black foo.py`.